### DISCUSSION TITLE
Initializing mlocate database; this may take some time...

### DISCUSSION THREAD

**Discussion by anaymalpani** [microsoft/WSL#5967] [General]
While I was trying to install kubuntu-desktop on wsl2(Ubuntu 20.04) on windows 10.

It got stuck at

> **Initializing mlocate database; this may take some time...**

after waiting for a long time(left it for a whole night, as I heard this process takes a lot of time as it parses through every path present on the system)

I closed the process and ran:

> **sudo dpkg --configure -a**

again it's stuck at the same point.

My question is:

How to resolve it?
Why does it take so much time?
Any help will be appreciated.

Thanks in advance!

**Comment by therealkenc** [comment]
Right answer. A a practical matter, `sudo apt remove mlocate` is another way to roll, unless you are actually dead set on using `locate(1)`.

Still wonder, idly, why "forgetting" to update `PRUNEPATHS` isn't disastrous across the whole Linux ecosystem (obviously, it isn't). I have a crazy slow USB spinny-disk drive with an NTFS filesystem on it, and I have never noticed a Real Linux box try to index it, even though it is ntfs-3g mounted automatically. Shrug. [edit, answering self] Looks like `/media` is pruned by default on Ubuntu, which is why it never comes up. But not `/mnt`. Which Ubuntu has never heard of.

![image](https://user-images.githubusercontent.com/7049700/93983860-f5609e80-fd37-11ea-876e-130df588dc33.png)

Another approach would be to add `9p` to `PRUNEFS`, along side analogous `NFS`, `afs`, `cifs`. [One supposes the maintainers didn't imagine anyone _actually using_ `9p`.]

Or.... an alternate-alternate approach would be to change [`/etc/wsl.conf`](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#automount) to use `[automount] root=/media`. Where God intended it.
